docs/fpc-registration-old.puml

```diff
@@ -0,0 +1,83 @@
+@startuml
+
+'- force wrapping of longer text
+'skinparam wrapWidth 400
+
+!pragma teoz true
+
+
+title Fabric Private Chaincode - Enclave registry - Current approach
+
+actor	Admin order 10
+participant ECC order 20
+participant ERCC order 40
+entity IAS order 50
+collections Orderer order 70
+collections Peers order 80
+
+note right of Peers
+  **Drawbacks**
+  - Custom validation at __every__ peer
+  - Single-endorser policy only
+  - ERCC must contact IAS (internect connection and IAS credentials needed)
+  - for EPID attestation SPID is needed when quote is created (ECC must provide this)
+end note
+
+group instantiation
+   Admin -> ECC : install
+   Admin -> ECC : instantiate (exp.mrenclave)
+   ECC -> ECC : putState(ecc.mrenclave, exp.mrenclave)
```

Comment:
what is exp.mrenclave? we have to somewhere also commit tlcc.mrenclave but i thought we would do that only during key-mgmt and so would be elsewhere (besides it would then also need a reference to the channel genesis block)